python/cudf/cudf/tests/dataset_synthesizer.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+# This module is for generating "synthetic" datasets. It was originally
+# designed for testing filtered reading. Generally, it should be useful
+# if you want to generate data where certain phenomena (e.g., cardinality)
+# are exaggurated.
+
+import pandas as pd
+import numpy as np
+
+from mimesis import Generic
+
+g = Generic("es")
+
+
+class ColumnParameters:
+    # There might be a case where we want to try out different distributions
+    # (i.e., something that isn't just uniform). For now, we just want to vary
+    # cardinality and presence of nulls in different columns.
+    #
+    # ty is a callable function, specifically a Mimesis provider
+    def __init__(
+        self,
+        cardinality=5,
+        null_frequency=0.1,
+        ty=g.food.fruit,
+        is_sorted=False,
+    ):
```

Comment:
Not sure if here is the right place, but can we allow specifying a seed into here generally to allow reproducible testing?